platform_plugin_aspects/static/js/embed_dashboard.js

```diff
@@ -1,11 +1,44 @@
-function embedDashboard(dashboard_uuid, superset_url, superset_token, xblock_id) {
+function embedDashboard(dashboard_uuid, superset_url, guest_token_url, xblock_id) {
   xblock_id = xblock_id || "";
+
+  function getCookie(name) {
+    let cookieValue = null;
+    if (document.cookie && document.cookie !== "") {
+      const cookies = document.cookie.split(";");
+      for (let i = 0; i < cookies.length; i++) {
+        const cookie = cookies[i].trim();
+        // Does this cookie string begin with the name we want?
+        if (cookie.substring(0, name.length + 1) === name + "=") {
+          cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+          break;
+        }
+      }
+    }
+    return cookieValue;
+  }
+
+  async function fetchGuestToken() {
+    return fetch(guest_token_url, {
+      method: 'POST',
+      headers: {
+        "X-CSRFToken": getCookie('csrftoken'),
+      },
+    }).then(response => {
+        if (response.ok) {
+          const data = response.json();
+          return data.guestToken;
+        } else {
+          console.error(response);
+        }
+    });
+  }
```

Comment:
This function isn't being called when I load the Instructor Dashboard filter, and I can't see why.. 
I'm also having issues with Superset returning 400 errors when I do hit it -- maybe dashboard UUIDs aren't matching up anymore.

Reply:
Do you have enabled all dashboard locales?

Reply:
The problem is that the function `fetchGuestToken` is an async function defined inside a normal function.